### PR TITLE
bugfix: Wrong span in extension methods in context functions

### PR DESCRIPTION
**Pull request by jkciesluk**
closes https://github.com/scalameta/metals/issues/5977
closes https://github.com/scalameta/metals/issues/5976

### CODE REVIEW

File: tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala [hunk 1]
```diff
@@ -385,4 +385,109 @@ class Scala3DocumentHighlightSuite extends BaseDocumentHighlightSuite {
        |""".stripMargin
   )
 
+  check(
+    "i5977",
```
Comment (tgodzik) [inline tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala:389]: Is this already fixed in the compiler? Why don't we need to ignore it?

**Review by tgodzik — APPROVED** [review]
Looks good, just one question

File: tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala (hunk 1 above)
Comment (jkciesluk) [reply tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala:389]: I think it was fixed by https://github.com/lampepfl/dotty/pull/18557

Comment (tgodzik) [reply tests/cross/src/test/scala/tests/highlight/Scala3DocumentHighlightSuite.scala:389]: Ach ok, thanks!

**Pull request merged**